x-pack/plugins/security_solution/public/common/components/guided_onboarding_tour/tour_step.tsx

```diff
@@ -150,6 +170,23 @@ export const GuidedOnboardingTourStep = ({
   // can be false if the anchor is an iterative element
   // do not use this as an "is tour active" check, the SecurityTourStep checks that anyway
   isTourAnchor = true,
+  hidden = false,
   ...props
-}: GuidedOnboardingTourStep) =>
-  isTourAnchor ? <SecurityTourStep {...props}>{children}</SecurityTourStep> : <>{children}</>;
+}: GuidedOnboardingTourStep) => {
+  const { cases } = useKibana().services;
+  const isAddToNewCaseFlyoutOpen = cases.hooks.useIsAddToNewCaseFlyoutOpen();
+  const isAddToExistingCaseModalOpen = cases.hooks.useAddToExistingCaseModalOpen();
+
+  const hiddenWhenCasesModalFlyoutExpanded = useMemo(
```

Comment:
To avoid this scenario:
<img width="2554" alt="Screenshot 2024-05-01 at 14 38 13" src="https://github.com/elastic/kibana/assets/6295984/39cd1c9b-5244-42f1-bfdb-3e6ced77b0b3">